### DISCUSSION TITLE
Lesson 3 - confused by all the zeros

### DISCUSSION THREAD

**Discussion by alkali333** [smartcontractkit/full-blockchain-solidity-course-py#1629] [Q&A] [answered]
Hi, 

I'm a little confused by the need to multiply by 10^18 in fund(), returning the answer * 10^10 in getPrice() and returning the rate / 10^18 in getConversion() rate. What is going on with all these 0s? I understand this is to use the Wei/Gwei standard, but I can't get my head around exactly why all these values are used. 

````
    function fund() public payable {
        uint256 minimumUSD = 50 * 10 ** 18;
        require(getConversionRate(msg.value) >= minimumUSD, "You need to spend more ETH");
        addressToAmountFunded[msg.sender] += msg.value;
    }


    function getVersion() public view returns(uint256) {
        AggregatorV3Interface priceFeed = AggregatorV3Interface(0x8A753747A1Fa494EC906cE90E9f37563A8AF630e);
        return priceFeed.version();
    }

    function getPrice() public view returns(uint256) {
        AggregatorV3Interface priceFeed = AggregatorV3Interface(0x8A753747A1Fa494EC906cE90E9f37563A8AF630e);
        (,int256 answer,,,) = priceFeed.latestRoundData();
        // ETH/USD Rate in 18 digit
        return uint256(answer * 10000000000);
    }

    function getConversionRate(uint256 ethAmount) public view returns (uint256) {
        uint256 ethPrice = getPrice(); //eth to usd price
        return (ethPrice * ethAmount) / 1000000000000000000;
    }`
```

**Comment by cromewar** [comment, marked as answer]
Hello @alkali333 
The reason Patrick made the code like this is just for standard reasons, you don't have to it as an obligation and the explanation is the following:
Most of ERC20 Tokens has a standard of 18 decimals, which is based on the WEI, GWEI and ETH conversion. so with that in mind:
`uint256 minimumUSD = 50 * 10 ** 18;` is just to fill up the standard.
`return uint256(answer * 10000000000);` the answer returns a 8 digit value, so the multiplication is again just to get 18.
`return (ethPrice * ethAmount) / 1000000000000000000` this is because eth price already have all the decimals so you don't want to have 36 decimals, just 18.

**Comment by alkali333** [comment]
Thanks for the explanation.  So basically, we always put things into 18 decimals. I think it is going to take me a while to familiar with the Wei, Gwei and ETH conversion. It seems rather easy to get the wrong number of 0s which is not ideal when it comes to sending money 🤣Get Outlook for Android
